src/main/java/uk/gov/justice/digital/delius/service/OffenderManagerService.java

```diff
@@ -203,10 +203,11 @@ private boolean isStaffInTeam(final Team team, final Staff staff) {
 
     @Transactional
     public void deallocatePrisonerOffenderManager(final String nomsNumber) {
-        final var offender = offenderRepository.findByNomsNumber(nomsNumber)
-                .orElseThrow(() -> new NotFoundException(format("Offender %s not found", nomsNumber)));
+        final var offender = offenderRepository.findMostLikelyByNomsNumber(nomsNumber)
+            .getOrElseThrow((e) -> e)
```

Comment:
this will then throw the duplicate exception and the controller advice will convert to a 409